wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java

```diff
@@ -228,9 +248,23 @@ void compile() throws IOException {
       }
     }
 
+    LinkedHashMap<ProtoType, TypeName> protoTypeToJavaType = new LinkedHashMap<>();
+    LinkedHashMap<ProtoType, TypeName> protoTypeToAdapterClass = new LinkedHashMap<>();
```

Comment:
You could save some work and change `WireCompiler` to use `Map<ProtoType, TypeName>` immediately instead of starting with `Map<String, String>` and then later converting.